### ISSUE TITLE
Different sequence length in one batch

### ISSUE THREAD

**Issue by MelodyVAR** [getao/icae#3]
I didn't see attention_mask in the forward, so if the batch length is inconsistent in the training process. How can I fix it.

**Comment by getao** [comment]
I don't think you need to set an attention mask because for the causal attention, padding tokens at the right of the sequences will not affect the training result.

**Comment by W-rudder** [comment]
> I don't think you need to set an attention mask because for the causal attention, padding tokens at the right of the sequences will not affect the training result.

The padding token is at the end of the sequence, but based on the provided code, the memory token is positioned to the right of the padding token. If the input sequence lengths are inconsistent, the padding token could affect the attention score of the memory token, right?
